Framework/src/Checker.cxx

```diff
@@ -70,120 +120,137 @@ Checker::~Checker()
+    //send(checkedMoArray, ctx.outputs());
+  }
 
   // monitoring
   endLastObject = system_clock::now();
-
-  // if 10 seconds elapsed publish stats
   if (timer.isTimeout()) {
     timer.reset(1000000); // 10 s.
     mCollector->send({ mTotalNumberHistosReceived, "objects" }, o2::monitoring::DerivedMetricMode::RATE);
   }
 }
 
-o2::header::DataDescription Checker::createCheckerDataDescription(const std::string taskName)
-{
-  if (taskName.empty()) {
-    BOOST_THROW_EXCEPTION(FatalException() << errinfo_details("Empty taskName for checker's data description"));
-  }
-  o2::header::DataDescription description;
-  description.runtimeInit(std::string(taskName.substr(0, o2::header::DataDescription::size - 4) + "-chk").c_str());
-  return description;
+void Checker::update(std::shared_ptr<MonitorObject> mo){
+  mLogger << mCheckerNames.front() << " - moMap key: " << mo->getTaskName() << AliceO2::InfoLogger::InfoLogger::endm;
+  mMonitorObjects[mo->getTaskName()] = mo;
+  mPolicy->update(mo->getTaskName());
 }
 
-void Checker::check(std::shared_ptr<MonitorObject> mo)
+std::vector<std::shared_ptr<QualityObject>> Checker::check(std::map<std::string, std::shared_ptr<MonitorObject>> moMap)
 {
-  std::map<std::string /*checkName*/, CheckDefinition> checks = mo->getChecks();
 
-  mLogger << "Running " << checks.size() << " checks for \"" << mo->getName() << "\""
+  mLogger << "Running " << mChecks.size() << " checks for " << moMap.size() << " monitor objects"
           << AliceO2::InfoLogger::InfoLogger::endm;
-  // Get the Checks
+  
+  std::vector<std::shared_ptr<QualityObject>> qualityVector;
 
   // Loop over the Checks and execute them followed by the beautification
-  for (const auto& [checkName, check] : checks) {
-    mLogger << "        check name : " << checkName << AliceO2::InfoLogger::InfoLogger::endm;
-    mLogger << "        check className : " << check.className << AliceO2::InfoLogger::InfoLogger::endm;
-    mLogger << "        check libraryName : " << check.libraryName << AliceO2::InfoLogger::InfoLogger::endm;
+  for (const auto& [checkName, checkInstance] : mChecks) {
+      mLogger << "        check name : " << checkName << AliceO2::InfoLogger::InfoLogger::endm;
+
+      // load module, instantiate, use check
+      // TODO : preload modules and pre-instantiate, or keep a cache
+      Quality q = checkInstance->check(&moMap);
 
-    // load module, instantiate, use check
-    // TODO : preload modules and pre-instantiate, or keep a cache
-    loadLibrary(check.libraryName);
-    CheckInterface* checkInstance = getCheck(checkName, check.className);
-    Quality q = checkInstance->check(mo.get());
+      mLogger << "  result of the check " << checkName << ": " << q.getName()
+              << AliceO2::InfoLogger::InfoLogger::endm;
 
-    mLogger << "  result of the check " << checkName << ": " << q.getName()
-            << AliceO2::InfoLogger::InfoLogger::endm;
+      if (mInputs.size() == 1 && moMap.size() == 1){
```

Comment:
It looks like this might be a problem. Which MOs from the input is taken is determined inside the Module, the checker does not know if the check was done on one or more MOs. An edge example to show the issue would be to specify many inputs for the checker, but use only one monitor object would be used in the check. The Checker cares only about the higher granularity, that is the InputSpec.

Reply:
Maybe the check declaration could contain more information on what monitor object it will use. With this, the issue could be easily solved, as we know how many MOs are used in the check. The implementation might be: 
- pass the map with all MOs (unsave/error prone - might use MO not declared in json)
- create every time new map with selected MOs
- create a shadowed map (class with interface of std::map, but pass list of available fields)
`
"checks": {
      "QcCheck": {
        "active": "true",
        "className": "o2::quality_control_modules::skeleton::SkeletonCheck",
        "moduleName": "QcSkeleton",
        "policy": "all",
        "dataSource": [{
          "type": "Task",
          "name": "QcTask",
          "MOs": [ "example" ]
        }]
      }
    }
`

Reply:
The 3rd option - not sure if I understand. A Check would be forbidden to access the MOs not specified in the list?
I would go for either 2nd or 3rd option. If the 3rd complicates the code significantly, but has little performance benefit, then maybe it is not worth it. Though there might exist tasks which produce hundreds or thousands of objects, then creating large containers frequently might pose a problem...
I like the idea of a list of MOs, but at the same time we should probably keep the possibility to order all MOs produced by a Task. What do you think would be a clear interface for that?
`"MOs": []` - an empty list is actually quite opposite to "all"
`"MOs": "all"` - this maybe clear, but requires checking if it is a list of object
`"MOs": [ "all" ]` - it can be mistaken for an MO name
` ` - maybe to order all MOs, we could just keep the "MOs" out.

Reply:
In the end, I choose the 2nd option for now - create every time a new map with selected MOs. Creating shadow map is not a problem and I was able to prepare such, but the problem is what interface should be implemented for a given shadow map.
As for the interface, I chose `"MOs": "all"`. It is considered default behaviour and will default to it in case of not specifying the MOs.